### PR TITLE
Add a method for testing - MobileCore.resetSDK()

### PR DESCRIPTION
**Pull request by yangyansong-adbe**

### CODE REVIEW

**Comment by praveek** [comment]
I don't recommend adding a public method to MobileCore. Can you change it to default accessor? Tests can access it by adding a helper class in com.adobe.marketing.mobile package.

**Comment by yangyansong-adbe** [comment]
https://developer.android.com/reference/androidx/annotation/VisibleForTesting
@praveek  It's only a public method in test code.

File: code/android-core-library/src/phone/java/com/adobe/marketing/mobile/MobileCore.java [hunk 1]
```diff
@@ -991,4 +991,10 @@ public static boolean dispatchResponseEvent(@NonNull final Event responseEvent,
         return true;
     }
 
+    @VisibleForTesting
+    public static void resetSDK() {
```
Comment (prudrabhat) [inline code/android-core-library/src/phone/java/com/adobe/marketing/mobile/MobileCore.java:995]: A public method visible for testing is a bit odd for an implementer using the SDK. What do you think about adding a test utility helper in the same package?

**Comment by yangyansong-adbe** [comment]
@praveek @prudrabhat addressed your review comments.

**Review by praveek — APPROVED** [review]

**Review by prudrabhat — APPROVED** [review]

File: code/android-core-library/src/phone/java/com/adobe/marketing/mobile/MobileCore.java [hunk 2]
```diff
@@ -991,4 +992,10 @@ public static boolean dispatchResponseEvent(@NonNull final Event responseEvent,
         return true;
     }
 
+    @VisibleForTesting
+    static void resetSDK() {
+        EventHub.Companion.setShared(new EventHub());
```
Comment (praveek) [inline code/android-core-library/src/phone/java/com/adobe/marketing/mobile/MobileCore.java:997]: Can you also shutdown the old event hub instance before setting a new one?

**Review by praveek — CHANGES_REQUESTED** [review]
Shutdown the event hub instance to avoid race conditions in test.

**Review by praveek — APPROVED** [review]

**Pull request merged**